File: post-comments/src/main/java/com/harsha/organize/entity/Post.java
package com.harsha.organize.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;


@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "post")
@Entity
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private int userId;
    private String name;
    private String body;
    private LocalDateTime createAt;
    private LocalDateTime updateAt= LocalDateTime.now();

    @OneToMany(mappedBy = "post")
    private List<Comment> comments;
}

File: post-comments 2/src/main/kotlin/com/harsha/organize/entity/Post.kt
package com.harsha.organize.entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime

@Table(name = "post")
data class Post(
    @Id
    var id: Int,
    @Column(value = "user_id")
    var userId: Int,
    var name: String = "",
    var body: String = "",
    var createAt: LocalDateTime,
    var updateAt: LocalDateTime = LocalDateTime.now()
)